core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -123,6 +123,24 @@ public fun CoroutineScope.peripheral(
     )
 }
 
+public fun CoroutineScope.peripheral(
+    macAddress: String,
+    builderAction: PeripheralBuilderAction = {},
+): Peripheral {
+    val builder = PeripheralBuilder()
+    builder.builderAction()
+    val bluetoothDevice = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(macAddress)
```

Comment:
While `getDefaultAdapter()` is guaranteed to return a `BluetoothDevice`, there's no guaranty that it can be connected to.

Android builds a cache of BluetoothDevices when scanning, and this cache can be cleared by various events (dependent on manufacture/OS version), so you can't reliably connect to a device with just a Mac Address.

When a device isn't cached, its `deviceType` returns `BluetoothDevice.DEVICE_TYPE_UNKNOWN`

Returning an `AndroidPeripheral` with a `bluetoothDevice` that can't connect is something we should avoid, though I'm not sure the best way to handle that.

Returning `null` as you do with the `CentralManager` implementation may be an option, though I'm leaning more towards surfacing a more actionable `BluetoothDeviceNotCachedException` error instead.

Definitely open to other suggestions though.

@twyatt is this something you've given any thought to?